pkg/oam/util/helper.go

```diff
@@ -988,3 +988,11 @@ func (accessor *applicationResourceNamespaceAccessor) Namespace() string {
 func NewApplicationResourceNamespaceAccessor(appNs, overrideNs string) NamespaceAccessor {
 	return &applicationResourceNamespaceAccessor{applicationNamespace: appNs, overrideNamespace: overrideNs}
 }
+
+// ComponentNameWithReplicaKey helps generate component name with replica key
+func ComponentNameWithReplicaKey(componentName, replicaKey string) string {
+	if replicaKey == "" {
+		return componentName
+	}
+	return fmt.Sprintf("%s-%s", componentName, replicaKey)
```

Comment:
what if a user define workload name in cue template?

Reply:
It should be verified if the old style will overwrite the workload name as well.